### DISCUSSION TITLE
v2.16.4

### DISCUSSION THREAD

**Discussion by helgoboss** [helgoboss/helgobox#1182] [Announcements]
- ReaLearn: [#1124](https://github.com/helgoboss/helgobox/issues/1124) Added possibility to do global MIDI transformation: See new options in target "MIDI: Send message" (yes!)
- ReaLearn: [#1158](https://github.com/helgoboss/helgobox/issues/1158) Added possibility to use keyboard input in addition to MIDI/OSC input (yes!)
- ReaLearn: [#1155](https://github.com/helgoboss/helgobox/issues/1155) Added menu entry "User interface => Background colors" to globally disable the use of background colors (useful when using custom Windows themes)
- ReaLearn: [#1157](https://github.com/helgoboss/helgobox/issues/1157) Improved control via keyboard by **not** processing keyboard input when user is entering text (needs REAPER 7.23+)
- ReaLearn: [#1170](https://github.com/helgoboss/helgobox/issues/1170) Fixed REAPER actions "ReaLearn: Find first mapping by source" (wouldn't find certain mappings) and "ReaLearn: Learn single mapping (reassigning source)" (always created new mappings)
- ReaLearn: [#1152](https://github.com/helgoboss/helgobox/issues/1152) Fixed occasional "FX not loadable" error (e.g. when removing FX)
- ReaLearn: [#1162](https://github.com/helgoboss/helgobox/issues/1162) Fixed timing information not being passed on when translating device MIDI to device MIDI via target "MIDI: Send message"
- ReaLearn: [#1126](https://github.com/helgoboss/helgobox/issues/1126) Fixed error "impossible" when on monitoring FX and pressing target menu button
- Playtime: [#1131](https://github.com/helgoboss/helgobox/issues/1131) Added the possibility to record clips from stopped state with a count-in/pre-roll (simply turn metronome on and record, right-click metronome to set length of pre-roll ... makes users who come from other session views feel more at home)
- Playtime: [#1163](https://github.com/helgoboss/helgobox/issues/1163) Added track PDC information, including help texts (as a warning not to expect miracles when using high-latency FX)
- Playtime: [#1168](https://github.com/helgoboss/helgobox/issues/1168) Added support for REAPER's "Smooth seek": Clips will keep playing as if nothing happened. Current limitation: Smooth-seek to the left still leads to retriggering when using PDC.
- Playtime: [#1119](https://github.com/helgoboss/helgobox/issues/1119) Improved usability by always playing ignited clips when starting one specific clip when playback is stopped (makes users who come from other session views feel more at home)
- Playtime: [#1146](https://github.com/helgoboss/helgobox/issues/1146) Improved section and fade behavior and defaults in order to prevent unwanted auto-fades
- Playtime: [#1132](https://github.com/helgoboss/helgobox/issues/1132) Improved usability by making the static section an expert option and renaming it to "Source cut"
- Playtime: [#1166](https://github.com/helgoboss/helgobox/issues/1166) Improved usability by replacing the "Silence mode" design with a more traditional "Stopped" design (also, Playtime now starts by default in stopped state)
- Playtime: [#1136](https://github.com/helgoboss/helgobox/issues/1136) Improved keyboard control by handling SPACE even if set as a **global** shortcut in REAPER (needs REAPER 7.23+)
- Playtime: [#1137](https://github.com/helgoboss/helgobox/issues/1137) Improved usability by letting SPACE keep ongoing recordings instead of discarding them
- Playtime: [#1164](https://github.com/helgoboss/helgobox/issues/1164) Fixed clips completely losing sync when enabling or disabling high-latency FX due to PDC (for now only fixed when REAPER transport is stopped)
- Playtime: [#1179](https://github.com/helgoboss/helgobox/issues/1179) Fixed audio clips playing back slightly delayed right after recording (bug was introduced in 2.16.2)
- Playtime: [#1143](https://github.com/helgoboss/helgobox/issues/1143) Fixed Launchpad Pro detection on Windows if device name has a prefix (e.g. '2- ')
- Playtime: [#1165](https://github.com/helgoboss/helgobox/issues/1165) Fixed inability to use MIDI input devices with IDs > 63
- Playtime: [#1149](https://github.com/helgoboss/helgobox/issues/1149) Fixed App not getting keyboard focus unless clicking on specific areas such as the matrix view (keyboard shortcuts now work immediately after opening the app)
- Playtime: [#1160](https://github.com/helgoboss/helgobox/issues/1160) Fixed play cursors in MIDI editor and waveform view noticeably off when using high-latency FX (due to PDC)
- Playtime: [#1156](https://github.com/helgoboss/helgobox/issues/1156) Fixed audio clips turning silent when starting REAPER playback and having high-latency FX in the signal flow (due to PDC)
- Playtime: [#1135](https://github.com/helgoboss/helgobox/issues/1135) Fixed tempo detection recordings not working via controller
- Playtime: [#1134](https://github.com/helgoboss/helgobox/issues/1134) Fixed immediate playback of clip in stopped state not working via controller

<hr /><em>This discussion was created from the release <a href='https://github.com/helgoboss/helgobox/releases/tag/v2.16.4'>v2.16.4</a>.</em>

**Comment by sdfalk** [comment]
Im getting an error (Detected 1 issue) that I should be running reaper 7.23 or Id have problems..Reaper 7.23 hasn't been released yet has it?

**Reply by helgoboss** [reply]
Yes. It's not an error, just a warning that not everything will work as it should (in this case it's about the handling of the SPACE key if you have defined a global shortcut for SPACE in REAPER).
